sfxclient/httpsink.go

```diff
@@ -240,59 +241,55 @@ func datapointAndEventResponseValidator(respBody []byte) error {
 	return nil
 }
 
-var toMTMap = map[datapoint.MetricType]com_signalfx_metrics_protobuf.MetricType{
-	datapoint.Counter:   com_signalfx_metrics_protobuf.MetricType_CUMULATIVE_COUNTER,
-	datapoint.Count:     com_signalfx_metrics_protobuf.MetricType_COUNTER,
-	datapoint.Enum:      com_signalfx_metrics_protobuf.MetricType_GAUGE,
-	datapoint.Gauge:     com_signalfx_metrics_protobuf.MetricType_GAUGE,
-	datapoint.Rate:      com_signalfx_metrics_protobuf.MetricType_GAUGE,
-	datapoint.Timestamp: com_signalfx_metrics_protobuf.MetricType_GAUGE,
+var toMTMap = map[datapoint.MetricType]sfxmodel.MetricType{
+	datapoint.Counter:   sfxmodel.MetricType_CUMULATIVE_COUNTER,
+	datapoint.Count:     sfxmodel.MetricType_COUNTER,
+	datapoint.Enum:      sfxmodel.MetricType_GAUGE,
+	datapoint.Gauge:     sfxmodel.MetricType_GAUGE,
+	datapoint.Rate:      sfxmodel.MetricType_GAUGE,
+	datapoint.Timestamp: sfxmodel.MetricType_GAUGE,
 }
 
-func toMT(mt datapoint.MetricType) com_signalfx_metrics_protobuf.MetricType {
+func toMT(mt datapoint.MetricType) sfxmodel.MetricType {
 	ret, exists := toMTMap[mt]
 	if exists {
 		return ret
 	}
 	panic(fmt.Sprintf("Unknown metric type: %d\n", mt))
 }
 
-func toEC(ec event.Category) com_signalfx_metrics_protobuf.EventCategory {
-	// Check if the event.Category does not have a corresponding com_signalfx_metrics_protobuf.EventCategory
-	if _, ok := com_signalfx_metrics_protobuf.EventCategory_name[int32(ec)]; !ok {
+func toEC(ec event.Category) sfxmodel.EventCategory {
+	// Check if the event.Category does not have a corresponding sfxmodel.EventCategory
+	if _, ok := sfxmodel.EventCategory_name[int32(ec)]; !ok {
 		panic(fmt.Sprintf("Unknown event category: %v\n", ec))
 	}
-	// Return the com_signalfx_metrics_protobuf.EventCategory
-	return com_signalfx_metrics_protobuf.EventCategory(int32(ec))
+	// Return the sfxmodel.EventCategory
+	return sfxmodel.EventCategory(ec)
 }
 
-func datumForPoint(pv datapoint.Value) *com_signalfx_metrics_protobuf.Datum {
+func datumForPoint(pv datapoint.Value) sfxmodel.Datum {
 	switch t := pv.(type) {
 	case datapoint.IntValue:
 		x := t.Int()
-		return &com_signalfx_metrics_protobuf.Datum{IntValue: &x}
+		return sfxmodel.Datum{IntValue: &x}
 	case datapoint.FloatValue:
 		x := t.Float()
-		return &com_signalfx_metrics_protobuf.Datum{DoubleValue: &x}
+		return sfxmodel.Datum{DoubleValue: &x}
 	default:
 		x := t.String()
-		return &com_signalfx_metrics_protobuf.Datum{StrValue: &x}
+		return sfxmodel.Datum{StrValue: &x}
 	}
 }
 
-func mapToDimensions(dimensions map[string]string) []*com_signalfx_metrics_protobuf.Dimension {
-	ret := make([]*com_signalfx_metrics_protobuf.Dimension, 0, len(dimensions))
+func mapToDimensions(dimensions map[string]string) []*sfxmodel.Dimension {
+	ret := make([]*sfxmodel.Dimension, 0, len(dimensions))
```

Comment:
Same trick can be applied to avoid unnecessary small allocations:

```
dims := make([]sfxmodel.Dimension, len(dimensions))
wrappers := make([]*sfxmodel.Dimension, len(dimensions))
ix := 0
for k, v := range dimensions {
	if k == "" || v == "" {
		continue
	}

	dims[ix].Key = filterSignalfxKey(k)
	dims[ix].Value = v
	wrappers[ix] := &dims[ix]
	ix++
}
return wrappers[:ix]
```